scripts/install_prerequesites_ubuntu_24_04.sh

```diff
@@ -1,20 +1,45 @@
 #!/bin/bash
 
-# Add Docker's official GPG key:
-sudo apt-get update -y
-sudo apt-get install ca-certificates curl -y
-sudo install -m 0755 -d /etc/apt/keyrings
-sudo curl -fsSL https://download.docker.com/linux/ubuntu/gpg -o /etc/apt/keyrings/docker.asc
-sudo chmod a+r /etc/apt/keyrings/docker.asc
+PIPENV_CMD="pipenv"
 
-# Add the docker repository to Apt sources:
-echo \
-    "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/ubuntu \
-  $(. /etc/os-release && echo "${UBUNTU_CODENAME:-$VERSION_CODENAME}") stable" |
-    sudo tee /etc/apt/sources.list.d/docker.list >/dev/null
-sudo apt-get update -y
+install_docker() {
+    # Add Docker's official GPG key:
+    sudo apt-get install ca-certificates curl -y
+    sudo install -m 0755 -d /etc/apt/keyrings
+    sudo curl -fsSL https://download.docker.com/linux/ubuntu/gpg -o /etc/apt/keyrings/docker.asc
+    sudo chmod a+r /etc/apt/keyrings/docker.asc
+
+    # Add the docker repository to Apt sources:
+    echo \
+        "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/ubuntu \
+    $(. /etc/os-release && echo "${UBUNTU_CODENAME:-$VERSION_CODENAME}") stable" |
+        sudo tee /etc/apt/sources.list.d/docker.list >/dev/null
+    sudo apt-get update -y
 
-sudo apt-get install -y docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin pipenv
+    sudo apt-get install -y docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
+}
+
+install_pipenv() {
+    if [ ${VERSION_ID} = "24.04" ]; then
+        sudo apt install -y pipenv
+    elif [ ${VERSION_ID} = "22.04" ]; then
+        curl -fsSL https://pyenv.run | bash
+        export PATH="${HOME}/.pyenv/bin:${HOME}/.pyenv/bin:${PATH}"
+        sudo apt-get install -y build-essential zlib1g-dev libffi-dev libssl-dev libbz2-dev libreadline-dev libsqlite3-dev liblzma-dev libncurses-dev tk-dev
+        sudo apt-get install -y python3-pip
+        pip3 install pipenv
+        export PIPENV_CMD="python3 -m pipenv"
+    fi
+}
+
+. /etc/os-release
+sudo apt-get update -y
+install_pipenv
+install_docker
+echo ${PIPENV_CMD}
+DOCKER_INSTALL_CMD="${PIPENV_CMD} run make docker-image; ${PIPENV_CMD} run make docker-image"
+SHELL_CMD="${PIPENV_CMD} shell"
 sudo usermod -a -G docker $(id -nu)
```

Comment:
please move this into `install_docker`

Reply:
install_docker, just creates docker it doesn't install the image, do you want a separate function that makes the image?